### PR TITLE
Use Typescript

### PR DESCRIPTION
**Pull request by jshakes**
Converts all functions to `.ts` and types the function parameters.

Had to rework module import/exports slightly to appease new Jest config

### CODE REVIEW

**Review by hallee — COMMENTED** [review]
Nice, I love to see more TypeScript!

How would you feel about actually using the compiler for this project? `qz-react` and `babel-typescript` is IMO a bit of a special situation due to the complexity of our build pipeline there, but here, nothing is stopping us from using `tsc` in the way it was intended. Namely, to build TypeScript files into plain JS. This would allow any JS project to use this package; as it’s currently set up, only TypeScript projects would be able to use this (and even then I think this needs special handling; typically TypeScript projects don't point the compiler to anything in node_modules)

This would involve splitting the project into `src/*.ts` and `build/*.js` directories and adding a tsconfig.json file, which we should do regardless.

File: .eslintrc [hunk 1]
```diff
@@ -1,5 +1,10 @@
 {
 	"extends": [
 		"@quartz/eslint-config-base"
+	],
+	"overrides": [
+		{
+			"files": [ "*.js", "*.ts" ],
+		}
```
Comment (hallee) [inline .eslintrc]: It doesn't look like this overrides is actually overriding anything

**Comment by jshakes** [comment]
> Nice, I love to see more TypeScript!
> 
> How would you feel about actually using the compiler for this project? `qz-react` and `babel-typescript` is IMO a bit of a special situation due to the complexity of our build pipeline there, but here, nothing is stopping us from using `tsc` in the way it was intended. Namely, to build TypeScript files into plain JS. This would allow any JS project to use this package; as it’s currently set up, only TypeScript projects would be able to use this (and even then I think this needs special handling; typically TypeScript projects don't point the compiler to anything in node_modules)
> 
> This would involve splitting the project into `src/*.ts` and `build/*.js` directories and adding a tsconfig.json file, which we should do regardless.

That sounds good to me, let's chat about it more at standup

File: .eslintrc (hunk 1 above)
Comment (jshakes) [reply .eslintrc]: Oh weird, I thought I had to include this in order for it to run on .ts files, but it seems to work without it

Comment (jshakes) [inline .eslintrc]:
```suggestion
```

**Review by hallee — APPROVED** [review]
Looks great!

**Pull request merged**